mllm/backends/ascend/AscendCommon.hpp

```
@@ -0,0 +1,133 @@
+  explicit AscendTimer(const char* tag, bool sync_before = true, bool sync_after = true);
+  ~AscendTimer();
+
+ private:
+  const char* tag_;
+  bool sync_before_;
+  bool sync_after_;
+  std::chrono::high_resolution_clock::time_point start_;
+};
+
+// Convenience macros for scoped timing.
+#define ASCEND_TIME_SCOPE(tag) ::mllm::ascend::AscendTimer timer_scope_##__LINE__(tag, true, true)
```

Clang (14.0.6)

[error] function-like macro 'ASCEND_TIME_SCOPE' used; consider a 'constexpr' template function

(cppcoreguidelines-macro-usage,-warnings-as-errors)